spec/workers/company_domain_worker_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe CompanyDomainWorker, sidekiq: true do
+  let!(:company) { create(:company, :worker_domain) }
+  let!(:worker) { CompanyDomainWorker.new }
+
+  before(:each) do
+    allow(worker).to receive(:link_open).and_return(response_html)
+
+    worker.perform(company.id)
+  end
+
+  describe 'with valid data' do
+    let!(:response_html) { Nokogiri::HTML(open('spec/factories/test.html')) }
+
+    it 'testing worker queueing' do
+      expect { CompanyDomainWorker.perform_async(1) }.to change(CompanyDomainWorker.jobs, :size).by(1)
```

Comment:
there's no need to test sidekiq
just test you own methods
instead, you may test if it's triggered in controller